chain/src/Pos/Chain/Block/Logic/Integrity.hs

```diff
@@ -201,50 +204,62 @@ verifyHeader pm VerifyHeaderParams {..} h =
                     -- a slot leader schedule as it would for the `OBFT ObftStrict`
                     -- and `Original` cases.
                     ObftLenientLeaders ldrs blkSecurityParam lastBlkSlots ->
-                        [  ( (blockSlotLeader `elem` ldrs)
-                            , sformat ("slot leader who published block, "%build%", is not an acceptable leader.")
-                                    blockSlotLeader)
-                            , ( (obftLeaderCanMint blockSlotLeader blkSecurityParam lastBlkSlots)
-                            , sformat ("slot leader who published block, "%build%", has minted too many blocks in the past "%build%" slots.")
+                        [   ( blockSlotLeader `elem` ldrs
+                            , sformat ("ObftLenient: slot leader who published block, "%build%", is not an acceptable leader.")
                                     blockSlotLeader
-                                    (getBlockCount blkSecurityParam))
-                            ]
+                              )
+                        ,   ( obftLeaderCanMint blockSlotLeader blkSecurityParam lastBlkSlots
+                            , sformat ("ObftLenient: slot leader who published block, "%build%", has minted too many blocks ("% build %") in the past "%build%" slots.")
+                                    blockSlotLeader
+                                    (blocksMintedByLeaderInLastKSlots blockSlotLeader $ getOldestFirst lastBlkSlots)
+                                    (getBlockCount blkSecurityParam)
+                            )
+                        ]
 
                     ObftStrictLeaders ldrs ->
-                        [  ( (Just blockSlotLeader == (scheduleSlotLeader ldrs))
-                            , sformat ("slot leader from schedule, "%build%", is different from slot leader who published block, "%build%".")
-                                    (scheduleSlotLeader ldrs)
-                                    blockSlotLeader)
-                            ]
+                        if isNothing (scheduleSlotLeader ldrs)
+                            then [ (isJust (scheduleSlotLeader ldrs), "ObftStrict: scheduled slot leader is missing") ]
```

Comment:
I believe we could just return a list consisting of both of these checks rather than doing an `if-then-else`
E.g.
```
[ (isJust (scheduleSlotLeader ldrs), "ObftStrict: scheduled slot leader is missing")
, ( Just blockSlotLeader == scheduleSlotLeader ldrs
  , sformat ("ObftStrict: slot leader from schedule, "%build%", is different from slot leader who published block, "%build%".")
            (scheduleSlotLeader ldrs)
            blockSlotLeader
  )
]
```
However, I think it's fine as is.

Reply:
If `scheduleSlotLeader ldrs` is `Nothing` then both of these errors would be presented so perhaps it's cleaner, presentation-wise, to use the `if-then-else`.